misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/engine/InvocationArgProcessor.java

```diff
@@ -72,34 +93,58 @@
         }
 
         Map<String, Value> argValues = new HashMap<>();
-        for (int i = 0; i < argEvaluators.size(); i++) {
-            Map.Entry<String, Evaluator> arg = argEvaluators.get(i);
+        for (int argIndex = 0, paramIndex = 0; argIndex < argEvaluators.size(); argIndex++) {
+            Map.Entry<String, Evaluator> arg = argEvaluators.get(argIndex);
             ArgType argType = getArgType(arg);
             if (argType == ArgType.POSITIONAL) {
                 if (namedArgsFound) {
                     throw new EvaluationException(String.format(EvaluationExceptionKind.CUSTOM_ERROR.getString(),
                             "positional args are not allowed after named args."));
-                } else if (restArgsFound) {
-                    throw new EvaluationException(String.format(EvaluationExceptionKind.CUSTOM_ERROR.getString(),
-                            "positional args are not allowed after rest args."));
                 }
 
-                if (remainingParams.isEmpty()) {
+                if (remainingParams.isEmpty() && !restArgsFound) {
                     throw new EvaluationException(String.format(EvaluationExceptionKind.CUSTOM_ERROR.getString(),
                             "too many arguments in call to '" + functionName + "'."));
                 }
 
-                String parameterName = params.get(i).getName().get();
-                Value argValue = arg.getValue().evaluate().getJdiValue();
-                // For the ballerina function parameters with type "any", the generated runtime method will only accept
-                // the args which are subtypes of "java.lang.Object". Therefore all the primitive typed arguments
-                // must be converted into their wrapper implementations (i.e. 'int' -> 'Integer'), before passing
-                // into the method.
-                if (params.get(i).typeDescriptor().typeKind().getName().equals(BVariableType.ANY.getString())) {
-                    argValue = getValueAsObject(context, argValue);
+                String parameterName = params.get(paramIndex).getName().get();
```

Comment:
Can we make parameterName a local variable inside the else block? #L144

Reply:
@prabushi changed in f131f53